docs/CONFIG-VARS.md

```diff
@@ -387,3 +388,101 @@ postgres_servers = {
   }
 }
 ```
+
+## Azure Application Gateway with Azure Web Application Firewall (WAF)
+
+Azure Web Application Firewall (WAF) on Azure Application Gateway provides centralized protection from common exploits and vulnerabilities. Please see the [sample input file](../examples/sample-input-app-gateway.tfvars) for Application Gateway creation example.
+
+| Name | Description | Type | Default | Notes |
+| :--- | :--- | :--- | :--- | :--- |
+| create_app_gateway | Creates Azure Application Gateway | bool | false | |
+| app_gateway_config | Map of Application Gateway configuration objects | map | null | All the variables that can be defined in the `app_gateway_config` are described in table below. |
+| waf_policy | A JSON file with all the WAF_Policy rules | map | null | The WAF policy has few required components see the details below. |
```

Comment:
Is it possible to have more than one of these? Asking as the samples that were shown seemed quite numerous and did not know if we could apply 1..n of these during a deployment.

Reply:
These variables here cannot be more than one. As of now, every deployment will only have one Application Gateway and with one WAF policy associated with it.
There are some parameters in the variable `app_gateway_config` that can have one or many objects.